### PR TITLE
VBM-127: add image type check

### PR DESCRIPTION
**Pull request by natata7**

### CODE REVIEW

File: packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts [hunk 1]
```diff
@@ -28,6 +40,14 @@ export const pickImageCamera = async (): Promise<Asset | null> => {
...
+  if (
+    response?.assets?.[0]?.type !== undefined &&
+    ALLOW_IMAGE_TYPES.indexOf(response?.assets?.[0]?.type)
+  ) {
+    notification.error(i18n.t('errors.WRONG_FORMAT'));
+
+    return null;
+  }
 
   return response?.assets?.[0] || null;
```
Comment (seal323) [inline packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts:52]: This part is the same for both functions  
```
if (response.errorCode && response.errorMessage) {
    notification.error(response.errorMessage, response.errorCode);
  }
   if (response.didCancel) {
     notification.info(i18n.t('common:errors.DID_CANCEL'));
   }
   if (
     response?.assets?.[0]?.type !== undefined &&
     ALLOW_IMAGE_TYPES.indexOf(response?.assets?.[0]?.type)
   ) {
     notification.error(i18n.t('errors.WRONG_FORMAT'));

     return null;
   }

   return response?.assets?.[0] || null;
```
I suggest moving it to the separate function, f.e `processImageRespose`

File: packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts [hunk 2]
```diff
@@ -15,6 +19,14 @@ export const pickImageLibrary = async (): Promise<Asset | null> => {
   if (response.didCancel) {
     notification.info(i18n.t('common:errors.DID_CANCEL'));
   }
+  if (
+    response?.assets?.[0]?.type !== undefined &&
+    ALLOW_IMAGE_TYPES.indexOf(response?.assets?.[0]?.type)
```
Comment (seal323) [inline packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts:24]: `ALLOW_IMAGE_TYPES.includes(response?.assets?.[0]?.type)` ?

**Review by AleksandrTermenzhy — APPROVED** [review]
Approved once **all conversations are resolved**

**Review by seal323 — APPROVED** [review]

File: packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts [hunk 3]
```diff
@@ -2,32 +2,42 @@ import {
...
-  const response = await launchImageLibrary(IMAGE_OPTIONS);
+const processImageResponse = (response: ImagePickerResponse) => {
   if (response.errorCode && response.errorMessage) {
     notification.error(response.errorMessage, response.errorCode);
   }
   if (response.didCancel) {
     notification.info(i18n.t('common:errors.DID_CANCEL'));
   }
+  if (response?.assets?.[0]?.type) {
+    if (!ALLOW_IMAGE_TYPES.includes(response?.assets?.[0]?.type.toString())) {
```
Comment (seal323) [inline packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts:23]: Can type be non string?

Comment (natata7) [reply packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts:23]: no

Comment (seal323) [reply packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts:24]: Then maybe we don't need to use ```type.toString()```

Comment (natata7) [reply packages/mobile/src/helpers/image/pick-image/pick-image.helper.ts]: updated

**Pull request merged**